wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextDsl.kt

```diff
@@ -0,0 +1,279 @@
+ *
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+public class TextDsl internal constructor(action: TextDsl.() -> Unit) {
+    init {
+        apply(action)
+    }
+
+    /**
+     * The text object being built.
+     *
+     * @author NoComment1105
+     */
+    @PublishedApi
+    internal val text: MutableText = Text.empty()
+
+    /**
+     * Adds a translatable text.
+     *
+     * @param value The translation key of the text.
+     * @param args Any arguments to apply to the translatable text. Can be left
+     * empty for no arguments.
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun translatable(
+        value: String,
+        vararg args: Any,
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.translatable(value, args)))
+    }
+
+    /**
+     * Adds a literal text.
+     *
+     * @param value The text.
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun literal(value: String, action: MutableStyle.() -> Unit = { }) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.literal(value)))
+    }
+
+    /**
+     * Adds a mutable key bind text.
+     *
+     * @param key The key of the Key bind
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun keyBind(key: String, action: MutableStyle.() -> Unit = { }) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.keyBind(key)))
+    }
+
+
+    /**
+     * @see Text.nbt
+     *
+     * @author NoComment1105
+     */
+    public inline fun nbt(
+        pathPattern: String,
+        interpreting: Boolean,
+        separator: Optional<Text>,
+        nbt: TextData,
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.nbt(pathPattern, interpreting, separator, nbt)))
+    }
+
+    /**
+     * @see Text.nbt
+     *
+     * @author NoComment1105
+     */
+    public inline fun nbt(
+        pathPattern: String,
+        interpreting: Boolean,
+        separator: Optional<Text>,
+        noinline nbt: ((ServerCommandSource) -> Stream<NbtCompound>),
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.nbt(pathPattern, interpreting, separator, nbt)))
+    }
+
+    /**
+     * Add a plain text.
+     *
+     * @param value The text to add
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public inline fun text(value: String, action: MutableStyle.() -> Unit = { }) {
```

Comment:
What's the difference of it from `literal`?

Reply:
Not overly sure, however the code provides both so we should mirror that imo